geode-core/build.gradle

```diff
@@ -163,93 +165,159 @@ dependencies {
-  compile('com.fasterxml.jackson.core:jackson-databind')
-  compile('commons-io:commons-io')
-  compile('commons-validator:commons-validator')
-  compile('commons-digester:commons-digester')
-  compile('com.sun.activation:javax.activation')
-  compile('javax.xml.bind:jaxb-api')
-  compile('com.sun.xml.bind:jaxb-impl')
-  compile('com.sun.istack:istack-commons-runtime') {
+
+  //Find bugs is used in multiple places in the code to suppress findbugs warnings
+  implementation('com.github.stephenc.findbugs:findbugs-annotations')
+
+  //Jgroups is a core component of our membership system.
+  implementation('org.jgroups:jgroups')
+
+  //Antlr is used by the query engine.
+  implementation('antlr:antlr')
+
+  //Jackson annotations is used in gfsh
+  implementation('com.fasterxml.jackson.core:jackson-annotations')
+
+  //Jackson databind is used in gfsh, and also in pdx
+  implementation('com.fasterxml.jackson.core:jackson-databind')
+
+  //Commons IO is used in persistence and management
+  implementation('commons-io:commons-io')
+
+  //Commons validator is used to validate inet addresses in membership
+  implementation('commons-validator:commons-validator')
+
+  //jaxb is used by cluster configuration
+  implementation('javax.xml.bind:jaxb-api')
+
+  //jaxb is used by cluster configuration
+  implementation('com.sun.xml.bind:jaxb-impl')
+
+  //istack appears to be used only by jaxb, not in our code. jaxb doesn't
+  //declare this as required dependency though. It's unclear if this is needed
+  //Runtime
+  runtimeOnly('com.sun.istack:istack-commons-runtime') {
     exclude group: '*'
   }
 
-  compile('org.apache.commons:commons-lang3')
-  compile('commons-modeler:commons-modeler') {
-    exclude module: 'commons-digester'
+  //Commons lang is used in many different places in core
+  implementation('org.apache.commons:commons-lang3')
+  
+  //Commons modeler is used by the (deprecated) admin API
+  implementation('commons-modeler:commons-modeler') {
     exclude module: 'commons-logging-api'
     exclude module: 'mx4j-jmx'
     exclude module: 'xml-apis'
     ext.optional = true
   }
-  compile('io.micrometer:micrometer-core')
     
-  compile('it.unimi.dsi:fastutil')
-  compile('javax.mail:javax.mail-api') {
+  //micrometer is used for micrometer based metrics from geode geode
+  api('io.micrometer:micrometer-core')
+
+
+  //FastUtil contains optimized collections that are used in multiple places in core
+  implementation('it.unimi.dsi:fastutil')
+
+  //Mail API is used by the deprecated admin API
+  implementation('javax.mail:javax.mail-api') {
     ext.optional = true
   }
-  compile('javax.resource:javax.resource-api')
-  compile('mx4j:mx4j') {
+
+  //The resource-API is used by the JCA support.
+  api('javax.resource:javax.resource-api')
+
+
+  //MX4J is used by the old admin API
+  implementation('mx4j:mx4j') {
     ext.optional = true
   }
-  compile('mx4j:mx4j-remote') {
+
+  //MX4J remote is used by both the old admin API 
```

Comment:
Fix the commnent. "Used by both the old admin API" and what?

Reply:
Nothing! Typo, I'll remove the both.

Reply:
Fixed!